zanata-war/src/main/java/org/zanata/service/impl/LocaleServiceImpl.java

```diff
@@ -263,8 +264,16 @@ public HLocale getByLocaleId(@Nonnull LocaleId locale) {
     }
 
     @Override
+    @Nullable
     public HLocale getByLocaleId(@Nonnull String localeId) {
-        return this.getByLocaleId(new LocaleId(localeId));
+        final LocaleId locale;
+        try {
+            locale = new LocaleId(localeId);
+        } catch (IllegalArgumentException e) {
+            log.warn("Tried to look up a locale with a malformed id", e);
+            return null;
+        }
+        return this.getByLocaleId(locale);
```

Comment:
This plus the variable declaration could live inside the `try {}` as a one liner.  (Not important though.)